pkg/credentialprovider/gcp/metadata.go

```diff
@@ -148,18 +163,74 @@ func (g *dockerConfigUrlKeyProvider) Provide() credentialprovider.DockerConfig {
 	return credentialprovider.DockerConfig{}
 }
 
+func runWithBackoff(url string, f func() ([]byte, error)) ([]byte, error) {
+	var backoff = time.Millisecond
+	const maxBackoff = time.Minute
+	for {
+		value, err := f()
+		if err == nil {
+			return value, nil
+		} else if httpErr, ok := err.(*credentialprovider.HttpError); ok {
+			if httpErr.StatusCode != http.StatusNotFound {
```

Comment:
As of now the logic is not retrying if the status code is not 404. Are you suggesting retrying for all error codes?